syntax = "proto3";

service DatasetFeed {
  
  rpc shutdown(Empty) returns (Empty) {}
  rpc registerjob(RegisterJobRequest) returns (RegisterJobResponse) {}
  rpc getBatches(GetBatchesRequest) returns (Sample) {}
}

message RegisterJobRequest{
    int32 job_id = 1;
  }

message GetBatchesRequest{
    int32 job_id = 1;
  }
message MessageResponse{
    string message = 1;
   }

message Empty {
}

message Sample {
   int32 batchid =1;
   //bytes image = 2;
   //bytes label = 3;
   string data = 2;
   string label = 3;
}

message RegisterJobResponse{
  string message = 1;
  bool successfully_registered = 2;
  int32 batches_per_epoch = 3;
  int32 dataset_length = 4;
 }
//cd sdl-server/protos
//python -m grpc_tools.protoc --proto_path=. ./data_feed.proto --python_out=.. --grpc_python_out=..